src/server/api/services/EventProcessor.ts

```diff
@@ -228,14 +231,41 @@ export class ObservationProcessor implements EventProcessor {
 
   async checkForAlerts(projectId: string, observationId: string) {
     try {
-      const totalTokenCost = await getTokenCostForUser(prisma, {
+      // Check for COST_PER_USER alerts
```

Comment:
Did you make a simple call or small load test to the server to check how much time we would spend with this logic on the new API route? I would like to ensure we do not add too much latency here.

Reply:
Yes I ran some local tests with a project with ~10k traces & 100k generations and it added about ~50ms in latency.